CSharpMath.CoreTests/LaTeXParserTest.cs

```diff
@@ -193,6 +193,32 @@ void RunScriptTest
       Assert.Equal(@"\sqrt[3]{2}", LaTeXParser.MathListToLaTeX(list).ToString());
     }
 
+    [Fact]
+    public void TestBra() {
+      var list = ParseLaTeX(@"\bra{i}");
+      Assert.Collection(list,
+        CheckAtom<Inner>("", inner => {
+          Assert.Equal("〈", inner.LeftBoundary.Nucleus);
+          Assert.Equal("|", inner.RightBoundary.Nucleus);
+          Assert.Collection(inner.InnerList, CheckAtom<Variable>("i"));
+        })
+      );
+      Assert.Equal(@"\bra{i}", LaTeXParser.MathListToLaTeX(list).ToString());
+    }
+
+    [Fact]
+    public void TestKet() {
+      var list = ParseLaTeX(@"\ket{i}");
+      Assert.Collection(list,
+        CheckAtom<Inner>("", inner => {
+          Assert.Equal("|", inner.LeftBoundary.Nucleus);
+          Assert.Equal("〉", inner.RightBoundary.Nucleus);
+          Assert.Collection(inner.InnerList, CheckAtom<Variable>("i"));
+        })
+      );
+      Assert.Equal(@"\ket{i}", LaTeXParser.MathListToLaTeX(list).ToString());
+    }
```

Comment:
Additional tests regarding parsing errors inside `\bra` and `\ket` should be added to `TestErrors` at the bottom of this file. The current implementation doesn't handle these cases correctly.

Reply:
Ok, I replicated some of the parsing error tests for powers and `\sqrt`.